js/views/inbox_view.js

```diff
@@ -6,34 +6,78 @@
 
     window.Whisper = window.Whisper || {};
 
-    var SocketView = Whisper.View.extend({
-        className: 'status',
+    Whisper.NetworkStatusView = Whisper.View.extend({
+        className: 'network-status',
         initialize: function() {
-            setInterval(this.updateStatus.bind(this), 5000);
-        },
-        updateStatus: function() {
-            var className, message = '';
-            if (typeof getSocketStatus === 'function') {
-              switch(getSocketStatus()) {
-                  case WebSocket.CONNECTING:
-                      className = 'connecting';
-                      break;
-                  case WebSocket.OPEN:
-                      className = 'open';
-                      break;
-                  case WebSocket.CLOSING:
-                      className = 'closing';
-                      break;
-                  case WebSocket.CLOSED:
-                      className = 'closed';
-                      message = i18n('disconnected');
-                      break;
-              }
-            if (!this.$el.hasClass(className)) {
-                this.$el.attr('class', className);
-                this.$el.text(message);
+            this.$el.hide();
+
+            setInterval(this.render.bind(this), 5000);
+            setTimeout(this.finishConnectingGracePeriod.bind(this), 5000);
```

Comment:
The idea is that it generally takes a second for the websocket to connect.  Ostensibly, this is not an error state, so no need to inform the user. When first opening the app, we hide network connectivity errors from the user for a grace period (5 seconds here).

Eliminating this grace period means a "Connecting" message and exclamation appears immediately when booting the app.  Which is actually kind of nice feedback, but maybe too much info for the average user.